packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -829,6 +829,15 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     setState(() {});
   }
 
+  void _onMenuClose() {
```

Comment:
@nate-thegrate Looks like this might address your comment from https://github.com/flutter/flutter/pull/156650#issuecomment-2408652577

so this might also fix https://github.com/flutter/flutter/issues/156570 ? (by letting the builder also rebuild when the menu closes)

Although, this is only for `DropdownMenu` and not `MenuAnchor`.

@bleroux is it perhaps possible to add this fix to `MenuAnchor` instead, and have `DropdownMenu` leverage it from there?

Reply:
@navaronbracke I read the great discussions on https://github.com/flutter/flutter/pull/156650.
At one point I even consider to use the listener approach, but later I realized we already have `MenuAnchor.onClose` and `MenuAnchor.onOpen` which are lighter than using a listener. So maybe we should consider closing https://github.com/flutter/flutter/issues/156570?

> @bleroux is it perhaps possible to add this fix to MenuAnchor instead, and have DropdownMenu leverage it from there?

This is not possible in that case because here the goal is to rebuild widgets that are outside the MenuAnchor (especially the trailing icon). Also @nate-thegrate comment pointed out that the builder issue was observed when the menu open not on close. We probably need another issue to track this.

Reply:
Yeah, I think we can close https://github.com/flutter/flutter/issues/156570 and file a follow-up issue for the bug when the menu opens.

Reply:
> This is not possible in that case because here the goal is to rebuild widgets that are outside of the MenuAnchor (especially the trailing icon).

It looks like the code prior to this PR had the trailing icon defined inside the MenuAnchor's builder. I'm wondering: if it wasn't rebuilding properly, maybe that could be fixed by addressing that menu anchor builder bug?

Reply:
In fact the the trailing icon was rebuilding properly previously. The goal of this PR is to move `_DropdownMenuBody` outside of the `MenuAnchor`. And, because `_DrowdownMenuBody` has the trailing icon as a child, I moved the trailing icon creation outside `MenuAnchor`.
I think we need more details for the menu anchor builder bug, can you file a separate issue with your findings?